File: python/python-broker/server/lib/parameter.py
#!/usr/bin/env python

class Parameter:
    key=None
    value=None

    is_raw_value=False

    def __init__(self, key, value, is_raw_value=False):
        self.key = key
        self.value = value

        self.is_raw_value = is_raw_value

    def getKey(self):
        return self.key

    def getValue(self):
        return self.value

    def getIsRawValue(self):
        return self.is_raw_value

File: python/python-broker/client/lib/parameter.php
<?php

/**
* @author PARMENTIER Laurent <lparmentier@quatral.com>
* @date 29-06-2015
*/

namespace Highshare\Conversion;

class Parameter {
	private $key;

	private $isRawValue;
	private $value;

	public function __construct($key, $value, $is_raw_value=false) {
		$this->key = $key;

		$this->is_raw_value = $is_raw_value;
		$this->value = $value;
	}

	public function getKey() {
		return $this->key;
	}

	public function getValue() {
		return $this->value;
	}

	public function appendValue($value) {
		$this->value .= $value;
	}

	public function isRawValue() {
		return $this->is_raw_value;
	}
}
